src/core/lib/channel/metrics.h

```diff
@@ -147,6 +193,18 @@ class StatsPlugin {
       GlobalInstrumentsRegistry::GlobalDoubleHistogramHandle handle,
       double value, absl::Span<const absl::string_view> label_values,
       absl::Span<const absl::string_view> optional_values) = 0;
+  virtual void SetGauge(
+      GlobalInstrumentsRegistry::GlobalUInt64GaugeHandle handle, uint64_t value,
+      absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) = 0;
+  virtual void SetGauge(
+      GlobalInstrumentsRegistry::GlobalDoubleGaugeHandle handle, double value,
+      absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) = 0;
+
+  virtual void AddCallback(RegisteredMetricCallback* callback) = 0;
+  virtual void RemoveCallback(RegisteredMetricCallback* callback) = 0;
```

Comment:
Don't we need to pass a certain type of instrument handle as an argument in this API? So that the StatsPlugin can find the async instrument to add the callback. We might be able to do something like:

```
using AsyncInstrumentHandle = absl::variant<GlobalInstrumentsRegistry::GlobalCallbackUInt64GaugeHandle, GlobalInstrumentsRegistry::GlobalCallbackDoubleGaugeHandle>;

virtual void AddCallback(AsyncInstrumentHandle handle, RegisteredMetricCallback* callback) = 0;
virtual void RemoveCallback(AsyncInstrumentHandle handle, RegisteredMetricCallback* callback) = 0;
```

If we want to minimize the API surface and to accommodate new async types in the future.

Reply:
The intent here is that a single callback can provide data for multiple metrics, so the callback is not associated with any one metric.

Consider a case like chttp2, where we will probably eventually have tens of metrics, all guarded by the same mutex.  In that case, we don't want to grab the lock separately for each individual metric, which is what we would need to do if we had a separate callback for each metric.  Instead, we want to be able to invoke a single callback that will grab the lock once and then report all of the metrics at once.

Reply:
I've changed the API to allow passing a list of metrics to be updated by each callback, along with a minimum interval at which the callback should be invoked.  These together will make it easier to implement caching in stats plugins like the OT plugin that do not support setting multiple metrics in a single callback.

Reply:
Added comments.